### PR TITLE
WFP-1325 setup db integration

### PR DESCRIPTION
**Pull request by carlov20**

### CODE REVIEW

File: docker-compose.yml [hunk 1]
```diff
@@ -1,17 +1,15 @@
 version: "3"
 services:
-  hmpps-probation-estate-api:
-    build:
-      context: .
-    network_mode: "host"
-    container_name: hmpps-probation-estate-api
+  postgres:
```
Comment (sp-watson) [inline docker-compose.yml:3]: Dunno if it is relevant but I usually put the postgres (and any other 3rd party tools) in a different docker compose - e.g. `docker-compose-test.yml`

File: helm_deploy/hmpps-probation-estate-api/values.yaml [hunk 2]
```diff
@@ -32,7 +32,11 @@ generic-service:
   namespace_secrets:
     hmpps-probation-estate-api:
       APPINSIGHTS_INSTRUMENTATIONKEY: "APPINSIGHTS_INSTRUMENTATIONKEY"
-
+    rds-instance-output:
```
Comment (sp-watson) [inline helm_deploy/hmpps-probation-estate-api/values.yaml:35]: Not sure if it is important, but I normally prefix the RDS instance secret with something, e.g. `pe-`

File: src/main/resources/application.yml [hunk 3]
```diff
@@ -7,7 +7,12 @@ spring:
     name: hmpps-probation-estate-api
   codec:
     max-in-memory-size: 10MB
-
+  r2dbc:
+    url: "r2dbc:postgresql://${database.endpoint:localhost:5432}/${database.name:postgres}"
+    username: "${database.username:root}"
+    password: "${database.password:dev}"
+  flyway:
+    locations: classpath:db/migration
```
Comment (sp-watson) [inline src/main/resources/application.yml:15]: This is the default, but I don't mind it been restated

**Review by sp-watson — APPROVED** [review]

File: docker-compose.yml (hunk 1 above)
Comment (markreesmoj) [reply docker-compose.yml:3]: What would the other docker-compose file be used for in that case?

File: .circleci/config.yml [hunk 4]
```diff
@@ -14,8 +14,9 @@ parameters:
 jobs:
   validate:
     executor:
-      name: hmpps/java
-      tag: "17.0"
+      name: hmpps/java_postgres
+      jdk_tag: "17.0"
```
Comment (markreesmoj) [inline .circleci/config.yml:18]: have we got 18 here yet?

File: src/main/resources/banner.txt [hunk 5]
```diff
@@ -1,13 +1,3 @@
...
- |  |___ |  | |    |___ |  |  |  |___ 
-                                      
-_  _ ____ ___ _    _ _  _ 
-|_/  |  |  |  |    | |\ | 
-| \_ |__|  |  |___ | | \| 
-                          
-Please change me by generating your own ASCII art and placing in banner.txt
+            __   __   __      __   __   __   __       ___    __           ___  __  ___      ___  ___
+|__|  |\/| |__) |__) /__`    |__) |__) /  \ |__)  /\   |  | /  \ |\ |    |__  /__`  |   /\   |  |__
+|  |  |  | |    |    .__/    |    |  \ \__/ |__) /~~\  |  | \__/ | \|    |___ .__/  |  /~~\  |  |___
```
Comment (markreesmoj) [inline src/main/resources/banner.txt:3]: 💟

**Review by sp-watson — APPROVED** [review]

File: docker-compose.yml (hunk 1 above)
Comment (sp-watson) [reply docker-compose.yml:3]: You mean the docker compose for the app itself? That's a good question - I suspect it's supposed to be used to run the whole service locally without having to set up the debugging tool.

I wonder why is it in the template?

**Pull request merged**